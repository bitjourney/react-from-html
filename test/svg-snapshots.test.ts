import "tsconfig-paths/register";
import renderer from "react-test-renderer";
import { ReactFromHtml } from "./entrypoint";
import fs from "fs";

describe("ReactFromHtml#parse", () => {
  const reactFromHtml = new ReactFromHtml();

  it("renders an SVG with style attributes", () => {
    const tree = renderer
      .create(
        reactFromHtml.parseToFragment(`
          <svg xmlns="http://www.w3.org/2000/svg" version="1.1" width="60" height="60">
            <rect width="60" height="60"/>
            <text text-anchor="middle" alignment-baseline="central" x="30" y="30" style="font-size: 15;fill: #327ac2;">Hello, world!</text>
          </svg>
        `)
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders a simple SVG", () => {
    // this SVG comes from https://developer.mozilla.org/en-US/docs/Web/SVG/Tutorial/Getting_Started
    const tree = renderer
      .create(
        reactFromHtml.parseToFragment(`
          <svg version="1.1"
            baseProfile="full"
            width="300" height="200"
            xmlns="http://www.w3.org/2000/svg">
            <rect width="100%" height="100%" fill="red" />
            <circle cx="150" cy="100" r="80" fill="green" />
            <text x="150" y="125" font-size="60" text-anchor="middle" fill="white">SVG</text>
          </svg>
          `)
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders an SVG with style attributes", () => {
    // this SVG comes from https://developer.mozilla.org/en-US/docs/Web/SVG/Tutorial/Other_content_in_SVG
    const tree = renderer
      .create(
        reactFromHtml.parseToFragment(`
          <svg version="1.1"
              xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink"
              width="200" height="200">
            <image x="90" y="-65" width="128" height="146" transform="rotate(45)"
                xlink:href="https://developer.mozilla.org/media/img/mdn-logo.png"/>
         </svg>
        `)
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  const svgDirname = `${__dirname}/../node_modules/payment-icons/svg/flat`;
  fs.readdirSync(svgDirname)
    .filter(basename => {
      return /\.svg$/.test(basename);
    })
    .forEach(basename => {
      it(`renders a real-world SVG (${basename}) copied from ${svgDirname}`, () => {
        const svg = fs.readFileSync(`${svgDirname}/${basename}`);
        const tree = renderer
          .create(reactFromHtml.parseToFragment(svg))
          .toJSON();
        expect(tree).toMatchSnapshot();
      });
    });
});
