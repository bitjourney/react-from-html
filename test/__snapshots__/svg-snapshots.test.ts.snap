// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReactFromHtml#parse renders a simple SVG 1`] = `
Array [
  "
          ",
  <svg
    baseProfile="full"
    height="200"
    version="1.1"
    width="300"
    xmlns="http://www.w3.org/2000/svg"
  >
    
            
    <rect
      fill="red"
      height="100%"
      width="100%"
    />
    
            
    <circle
      cx="150"
      cy="100"
      fill="green"
      r="80"
    />
    
            
    <text
      fill="white"
      fontSize="60"
      textAnchor="middle"
      x="150"
      y="125"
    >
      SVG
    </text>
    
          
  </svg>,
  "
          ",
]
`;

exports[`ReactFromHtml#parse renders an SVG with style attributes 1`] = `
Array [
  "
          ",
  <svg
    height="60"
    version="1.1"
    width="60"
    xmlns="http://www.w3.org/2000/svg"
  >
    
            
    <rect
      height="60"
      width="60"
    />
    
            
    <text
      alignmentBaseline="central"
      style={Object {}}
      textAnchor="middle"
      x="30"
      y="30"
    >
      Hello, world!
    </text>
    
          
  </svg>,
  "
        ",
]
`;

exports[`ReactFromHtml#parse renders an SVG with style attributes 2`] = `
Array [
  "
          ",
  <svg
    height="200"
    version="1.1"
    width="200"
    xmlns="http://www.w3.org/2000/svg"
    xmlnsXlink="http://www.w3.org/1999/xlink"
  >
    
            
    <image
      height="146"
      transform="rotate(45)"
      width="128"
      x="90"
      xlinkHref="https://developer.mozilla.org/media/img/mdn-logo.png"
      y="-65"
    />
    
         
  </svg>,
  "
        ",
]
`;
